consul/Dockerfile

```
@@ -10,7 +10,10 @@ WORKDIR /work
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 RUN git clone --depth=1 -b v${CONSUL_VERSION} https://github.com/hashicorp/consul /work/consul
 
-RUN curl -sSfL -o docker-entrypoint.sh https://raw.githubusercontent.com/hashicorp/docker-consul/${DOCKER_CONSUL_REVISION}/0.X/docker-entrypoint.sh \
+# docker-entrypoint.sh contains non- bourne shell compatible variable substitution
+RUN curl -sSfL -o docker-entrypoint.sh.orig https://raw.githubusercontent.com/hashicorp/docker-consul/${DOCKER_CONSUL_REVISION}/0.X/docker-entrypoint.sh \
+    && head -n1  docker-entrypoint.sh.orig | sed 's/\/bin\/sh/\/bin\/bash/' > docker-entrypoint.sh \
+    && tail -n+2 docker-entrypoint.sh.orig >> docker-entrypoint.sh \
```

It would be good for readability to use a different delimiter from `/` when processing file paths.
You can specify an address `1` in front of `s` to substitute the 1st line only.

```suggestion
RUN curl -sSfL -o docker-entrypoint.sh https://raw.githubusercontent.com/hashicorp/docker-consul/${DOCKER_CONSUL_REVISION}/0.X/docker-entrypoint.sh \
    && sed -i -e '1s,/bin/sh,/bin/bash,' docker-entrypoint.sh \
```